lib/modules/datasource/helm/index.ts

```diff
@@ -34,63 +32,22 @@ export class HelmDatasource extends Datasource {
 
   @cache({
     namespace: `datasource-${HelmDatasource.id}`,
-    key: (helmRepository: string) => helmRepository,
+    key: (helmRepository: string) => `repository-data:${helmRepository}`,
   })
-  async getRepositoryData(
-    helmRepository: string,
-  ): Promise<HelmRepositoryData | null> {
-    let res: HttpResponse<string>;
-    try {
-      res = await this.http.get('index.yaml', {
+  async getRepositoryData(helmRepository: string): Promise<HelmRepositoryData> {
+    const { val, err } = await Result.wrap(
+      this.http.get('index.yaml', {
```

Comment:
we should add a `Http.getYaml` function similar to the `getJson` , so we can simply it further